tests/utils/execution.hpp

```diff
@@ -227,3 +233,68 @@ void kernel_run(auto nd_range, auto validate_result) {
     free(B_host);
     free(C_host);
 }
+
+/// @brief Using gpu_arch of current machine to run F<arch>::exec
+///
+/// @tparam F The gpu_arch-templated function wrapper
+///
+/// @example example usage in /examples/01 or /examples/02
+template <template <gpu_arch> class F>
+class dispatch_arch {
+    using T_RET = std::invoke_result_t<decltype(F<gpu_arch::Xe>::exec)>;
+
+public:
+    template <typename... Args>
+    static T_RET exec(Args &&...args) {
+        sycl::device device;
+        if (!device.has(aspect::ext_intel_device_id))
+            throw std::runtime_error("Can not get device ID");
+        auto deviceID = device.get_info<ext::intel::info::device::device_id>();
+        std::cout << "deviceID: 0x" << std::hex //
+                  << std::right << std::setfill('0') << deviceID << "\n";
+#if defined(SYCL_EXT_ONEAPI_DEVICE_ARCHITECTURE) \
+        && SYCL_EXT_ONEAPI_DEVICE_ARCHITECTURE
+        // https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc#feature-test-macro
+        namespace ENS = sycl::ext::oneapi::experimental;
+        auto deviceArch = device.get_info<ENS::info::device::architecture>();
+        switch (deviceArch) {
+            case ENS::architecture::intel_gpu_pvc:
+                return F<gpu_arch::Xe>::exec(std::forward<Args>(args)...);
+                return;
+            case ENS::architecture::intel_gpu_dg2_g10:
+            case ENS::architecture::intel_gpu_dg2_g11:
+            case ENS::architecture::intel_gpu_dg2_g12:
+                return F<gpu_arch::Dg2>::exec(std::forward<Args>(args)...);
+            default: break;
```

Comment:
seems no meteorlake support, can you raise an issue to them?

Reply:
Good idea

Reply:
https://github.com/intel/llvm/issues/12988